Uncaught AssertionError when using `$this` inside a conditional in non-class scope

```php
<?php

function run() {
	if ( rand() ) {
		$this->addFields();
	}
}
```

Phan crashes with something like:

```
Caught AssertionError: Cannot get class FQSEN on scope in phar:///phan-v6-dev.phar/src/Phan/Language/Scope/GlobalScope.php:226Stack trace:#0 phar:///phan-v6-dev.phar/src/Phan/Language/Scope/FunctionLikeScope.php(34): Phan\Language\Scope\GlobalScope->getClassFQSEN()#1 phar:///phan-v6-dev.phar/src/Phan/Language/Context.php(540): Phan\Language\Scope\FunctionLikeScope->getClassFQSEN()#2 phar:///phan-v6-dev.phar/src/Phan/Analysis/BlockExitStatusChecker.php(806): Phan\Language\Context->getClassFQSEN()#3 phar:///phan-v6-dev.phar/src/Phan/Analysis/BlockExitStatusChecker.php(591): Phan\Analysis\BlockExitStatusChecker->computeStatusOfMethodCall(Object(ast\Node))#4 phar:///phan-v6-dev.phar/src/Phan/AST/Visitor/KindVisitorImplementation.php(35): Phan\Analysis\BlockExitStatusChecker->visitMethodCall(Object(ast\Node))#5 phar:///phan-v6-dev.phar/src/Phan/Analysis/BlockExitStatusChecker.php(108): Phan\AST\Visitor\KindVisitorImplementation->__invoke(Object(ast\Node))#6 phar:///phan-v6-dev.phar/src/Phan/Analysis/BlockExitStatusChecker.php(1000): Phan\Analysis\BlockExitStatusChecker->check(Object(ast\Node))#7 phar:///phan-v6-dev.phar/src/Phan/Analysis/BlockExitStatusChecker.php(873): Phan\Analysis\BlockExitStatusChecker->computeStatusOfBlock(Array)#8 phar:///phan-v6-dev.phar/src/Phan/AST/Visitor/KindVisitorImplementation.php(35): Phan\Analysis\BlockExitStatusChecker->visitStmtList(Object(ast\Node))#9 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(2343): Phan\AST\Visitor\KindVisitorImplementation->__invoke(Object(ast\Node))#10 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(648): Phan\BlockAnalysisVisitor->visitIf(Object(ast\Node))#11 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(242): Phan\BlockAnalysisVisitor->analyzeAndGetUpdatedContext(Object(Phan\Language\Context), Object(ast\Node), Object(ast\Node))#12 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(648): Phan\BlockAnalysisVisitor->visitStmtList(Object(ast\Node))#13 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(3269): Phan\BlockAnalysisVisitor->analyzeAndGetUpdatedContext(Object(Phan\Language\Context), Object(ast\Node), Object(ast\Node))#14 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(3290): Phan\BlockAnalysisVisitor->visitMethod(Object(ast\Node))#15 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(648): Phan\BlockAnalysisVisitor->visitFuncDecl(Object(ast\Node))#16 phar:///phan-v6-dev.phar/src/Phan/BlockAnalysisVisitor.php(242): Phan\BlockAnalysisVisitor->analyzeAndGetUpdatedContext(Object(Phan\Language\Context), Object(ast\Node), Object(ast\Node))#17 phar:///phan-v6-dev.phar/src/Phan/AST/Visitor/KindVisitorImplementation.php(35): Phan\BlockAnalysisVisitor->visitStmtList(Object(ast\Node))#18 phar:///phan-v6-dev.phar/src/Phan/Analysis.php(806): Phan\AST\Visitor\KindVisitorImplementation->__invoke(Object(ast\Node))#19 phar:///phan-v6-dev.phar/src/Phan/Phan.php(620): Phan\Analysis::analyzeFile(Object(Phan\CodeBase), 'input', NULL, NULL)#20 phar:///phan-v6-dev.phar/src/Phan/Phan.php(677): Phan\Phan::{closure:Phan\Phan::finishAnalyzingRemainingStatements():618}(0, 'input', 1)#21 phar:///phan-v6-dev.phar/src/Phan/Phan.php(468): Phan\Phan::finishAnalyzingRemainingStatements(Object(Phan\CodeBase), NULL, Array, Array, NULL)#22 PIB(85): Phan\Phan::analyzeFileList(Object(Phan\CodeBase), Object(Closure))#23 {main}
```

([Demo](https://phan.github.io/demo/?c=DwfgDgFmBQ0GYFcB2BjALgSwPZIAQCdkAKASlwG9oBIDOXIggQyQBNTczKqqASNCDAGcAtAD5GLFgDEMAUwA2LQaQDc1AL7RNQA&php=84&phan=v6-dev&ast=1.1.3&plugins=310592423266))

--- a/src/Phan/Analysis/BlockExitStatusChecker.php
+++ b/src/Phan/Analysis/BlockExitStatusChecker.php
@@ -803,7 +803,10 @@ final class BlockExitStatusChecker extends KindVisitorImplementation
             $method_name = $method;
         }
         if ($var_name === 'this') {
-            $class_fqsen = $this->context->getClassFQSEN();
+            $class_fqsen = $this->context->getClassFQSENOrNull();
+            if ($class_fqsen === null) {
+                return self::STATUS_PROCEED;
+            }
         } else {
             // TODO not yet handled
             return self::STATUS_PROCEED;

--- a/tests/files/expected/nonclass_this_in_conditional.php.expected
+++ b/tests/files/expected/nonclass_this_in_conditional.php.expected
@@ -1,0 +1,1 @@
+%s:4 PhanUndeclaredThis Variable $this is undeclared

--- a/tests/files/src/nonclass_this_in_conditional.php
+++ b/tests/files/src/nonclass_this_in_conditional.php
@@ -1,0 +1,6 @@
+<?php
+function run() {
+    if (rand()) {
+        $this->addFields();
+    }
+}

